### ISSUE TITLE
cli_parse module crashing with PyATS Parser (Python 3 on OS X Catalina)

### ISSUE THREAD

**Issue by sswitalski18** [ansible-collections/ansible.netcommon#164]
##### SUMMARY
Attempting to parse Cisco Nexus device configuration with the `cli_parse` module.  A playbook configured to use the native parser runs to completion successfully.  However, switching the parser argument to `ansible.netcommon.pyats` causes an issue.

According to the `cli_parse` documentation, PyATS, Genie and Python 3 are required for this module to work.  I've installed these pre-requisites with pip both in my native Python 3 environment and a virtual environment.

I'm launching the following playbook from my host, OS X running 10.15.5

```
- name: grab info
  ansible.netcommon.cli_parse:
    command: show interface
    parser:
      name: ansible.netcommon.pyats
    set_fact: interfaces

- name: push structured data to file
  copy:
    content: "{{ interfaces | to_nice_yaml }}"
    dest: "{{ playbook_dir }}/roles/nxos_rm_testing/config/{{ inventory_hostname }}"
```

Python 3 will then crash with the following errors:
```
objc[11536]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[11536]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
ERROR! A worker was found in a dead state
```

Not positive if this is a module error or a PyATS/Genie error but I figured I'd start my journey here.  Has anyone in the community experienced this issue?  Am I missing something in the playbook? Do I need to be running a specific PyATS/Genie combination?

Thanks!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible.netcommon.cli_parse

##### ANSIBLE VERSION
```
ansible 2.10.2
  config file = /Users/home_dir/.ansible.cfg
  configured module search path = ['/Users/home_dir/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/home_dir/Library/Python/3.8/lib/python/site-packages/ansible
  executable location = /Users/home_dir/Library/Python/3.8/bin/ansible
  python version = 3.8.2 (default, Aug 25 2020, 09:23:57) [Clang 12.0.0 (clang-1200.0.32.2)]
```

##### CONFIGURATION
```
<null>
```

##### OS / ENVIRONMENT
Ansible Core Host: OS X 10.15.5, Python 3
Target: Cisco Nexus 9000, 9.3.3

##### STEPS TO REPRODUCE
Running the following playbook will cause the Python environment to crash and generate a traceback

```
- name: grab info
  ansible.netcommon.cli_parse:
    command: show interface
    parser:
      name: ansible.netcommon.pyats
    set_fact: interfaces

- name: push structured data to file
  copy:
    content: "{{ interfaces | to_nice_yaml }}"
    dest: "{{ playbook_dir }}/roles/nxos_rm_testing/config/{{ inventory_hostname }}"
```

```
objc[11536]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[11536]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
ERROR! A worker was found in a dead state
```

##### EXPECTED RESULTS
Python should not crash and the playbook should run to completion.

##### ACTUAL RESULTS
Python crashes with the following traceback:
```
objc[11536]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[11536]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
ERROR! A worker was found in a dead state
```

**Comment by sswitalski18** [comment]
Since opening this ticket I've confirmed that I can run the `show interface` parsing command from PyATS itself.  This leads me to believe there might be an issue with Ansible's execution of the PyATS command.

One thing of note.  As I'm also attempting to learn PyATS/Genie, It seems that the PyATS testbed requires the device hostname to align with the hostname in the testbed yaml file.  I've since updated my ansible inventory so that the hostname is identical to the hostname on the nexus device and I still encounter the same crash/traceback.  Previously I just had the IP address in my inventory file.

**Comment by sswitalski18** [comment]
Found an interesting development.  I ran into the same issue with the netbox plugin.  This means to me it's a platform/environment issue and not this particular plugin.

I searched for this error more generically outside of ansible and came across a ruby thread where users were encountering the same issue.

I'm now able to get past this initial issue by adding the following to my environment

```export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES```

I'll be closing this issue as a result.

Thanks!
